zero_bin/rpc/src/main.rs

```diff
@@ -1,94 +1,168 @@
-                    // Get the prover input for particular block.
-                    let result = rpc::block_prover_input(
-                        cached_provider.clone(),
-                        block_id,
-                        checkpoint_state_trie_root,
-                        rpc_type,
-                    )
-                    .await?;
+        let block_interval =
+            BlockInterval::Range(self.params.start_block..self.params.end_block + 1);
 
-                    block_prover_inputs.push(result);
-                }
+        let cached_provider = Arc::new(CachedProvider::new(build_http_retry_provider(
+            self.params.rpc_url.clone(),
+            self.params.backoff,
+            self.params.max_retries,
+        )));
 
+        match self.command {
+            Command::Fetch {} => {
+                let block_prover_inputs =
+                    retrieve_block_prover_inputs(cached_provider, block_interval, &self.params)
+                        .await?;
                 serde_json::to_writer_pretty(std::io::stdout(), &block_prover_inputs)?;
             }
+            Command::Extract { tx } => {
+                let tx_hash: B256 = tx.parse()?;
+                let block_prover_inputs = retrieve_block_prover_inputs(
+                    cached_provider.clone(),
+                    block_interval.clone(),
+                    &self.params,
+                )
+                .await?;
+                let mut extracted_generation_input: Option<GenerationInputs> = None;
+
+                // Filter transactions
+                for block_prover_input in block_prover_inputs {
+                    let block_number: u64 = block_prover_input.get_block_number().try_into()?;
+                    let block = cached_provider
+                        .get_block(
+                            BlockId::Number(BlockNumberOrTag::Number(block_number)),
+                            BlockTransactionsKind::Hashes,
+                        )
+                        .await?;
+                    let generation_inputs = trace_decoder::entrypoint(
+                        block_prover_input.block_trace,
+                        block_prover_input.other_data,
+                        |_| unimplemented!(),
+                    )?;
+
+                    if let Some((index, _)) = block
+                        .transactions
+                        .hashes()
+                        .enumerate()
+                        .find(|it| *it.1 == tx_hash)
+                    {
+                        extracted_generation_input = generation_inputs.get(index).cloned();
+                    }
+                }
+                if extracted_generation_input.is_some() {
+                    serde_json::to_writer(std::io::stdout(), &extracted_generation_input)?;
+                } else {
+                    println!(
+                        "Transaction {} not found in block interval {}",
```

Comment:
Suggestion: How about looking up the block number here through the endpoint and helping the user?

    Warning: txn 0xC0FFEE exists in block 42 which is outside the requested interval [123,9000].
    Now repeating the command with `--start-block=42 --end-block=42`...

Reply:
Agreed that'd be cleaner